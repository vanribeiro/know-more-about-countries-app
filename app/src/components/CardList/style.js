import styled from "styled-components";

const CountryCardsSection = styled.section`
  width: 100%;
  margin-top: 50px;
  display: grid;
  row-gap: 50px;
  justify-content: space-between;

  @media screen and (max-width: 539px) {
    grid-template-columns: repeat(1, auto);
    justify-content: center;
  }

  @media screen and (min-width: 540px) {
    grid-template-columns: repeat(2, 48%);
  }

  @media screen and (min-width: 768px) {
    grid-template-columns: repeat(3, 30%);
  }

  @media screen and (min-width: 1200px) {
    grid-template-columns: repeat(4, 22%);
  }

  @media screen and (min-width: 1300px) {
    grid-template-columns: repeat(4, 21%);
  }

  @media screen and (min-width: 1400px) {
    grid-template-columns: repeat(4, 20%);
  }
`;

export {
    CountryCardsSection
}