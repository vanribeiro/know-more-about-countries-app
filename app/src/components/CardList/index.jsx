import Filters from "../../components/Filters";
import CountryCard from "../../components/CountryCard";
import { useState } from "react";
import { Link } from "react-router-dom";
import { NotFoundCountry } from "../MessageStatus";
import { CountryCardsSection } from './style';

const CardList = ({ cardsInfo }) => {
  const [nameSearched, setNameSearched] = useState("");
  const [clearInput, setClearInput] = useState(false);

  const onSearchSubmit = (text) => text !== "" ? setNameSearched(text) : setNameSearched("");

  const countrySearch = (cardsInfo) => {
    return cardsInfo.filter((cardInfo) => {
      if (cardInfo.name.common.includes(nameSearched)) {
        return cardInfo.name.common.includes(nameSearched);
      } else if (clearInput) {
        return cardsInfo;
      }
      return "";
    });
  };

  return (
    <>
      <Filters
        onSearchSubmit={onSearchSubmit}
        setClearInput={setClearInput}
        clearInput={clearInput}
      />
      <>
        {countrySearch(cardsInfo).length === 0 ? <NotFoundCountry /> : ""}
        <CountryCardsSection>
          {countrySearch(cardsInfo).map((info, id) => {
            return (
              <Link
                key={id}
                to={`/country/${info.name.common
                  .replaceAll(/\s/ig, "%20")
                  .replaceAll(/\(/ig, "%28")
                  .replaceAll(/\)/ig, "%29")
                  .toLowerCase()}`}
              >
                <CountryCard info={info} />
              </Link>
            );
          })}
        </CountryCardsSection>
      </>
    </>
  );
};

export default CardList;
