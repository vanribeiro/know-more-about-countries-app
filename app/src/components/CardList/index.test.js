/* eslint-disable testing-library/no-node-access */
import React from 'react';
import CardList from ".";
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

describe('CardList component', () => {
    it('should always have the same snapshot', () => {
        const cardsInfo = [{
            flags: {
              png: "https://flagcdn.com/w320/me.png",
              svg: "https://flagcdn.com/me.svg"
            },
            name: {
              common: "Montenegro",
            },
            capital: [
              "Podgorica"
            ],
            region: "Europe",
            population: 621718
        }];
        
        const { container } = render(
          <BrowserRouter>
            <CardList cardsInfo={cardsInfo}/>
          </BrowserRouter>
        );

        // eslint-disable-next-line no-undef
        // eslint-disable-next-line testing-library/no-container
        expect(container.querySelector('section')).toMatchSnapshot();
    });
});