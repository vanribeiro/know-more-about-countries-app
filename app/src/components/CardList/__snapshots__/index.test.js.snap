// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component should always have the same snapshot 1`] = `
<section
  class="sc-dJjYzT FeLIa"
>
  <a
    href="/country/montenegro"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-egiyK hHYkvZ css-66l59o-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="Montenegro Flag"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img sc-bqiRlB jRfFxj css-o69gx8-MuiCardMedia-root"
        src="https://flagcdn.com/me.svg"
      />
      <div
        class="MuiCardContent-root sc-hBUSln fsJbhJ css-46bh2p-MuiCardContent-root"
      >
        <h2
          class="sc-ksdxgE jdkhdz"
        >
          Montenegro
        </h2>
        <p
          class="sc-jrQzAO ggsRfk"
        >
          <strong
            class="sc-pVTFL kDtcZD"
          >
            Population: 
          </strong>
          621,718
        </p>
        <p
          class="sc-jrQzAO ggsRfk"
        >
          <strong
            class="sc-pVTFL kDtcZD"
          >
            Region: 
          </strong>
          Europe
        </p>
        <p
          class="sc-jrQzAO ggsRfk"
        >
          <strong
            class="sc-pVTFL kDtcZD"
          >
            Capital: 
          </strong>
          Podgorica
        </p>
      </div>
    </div>
  </a>
</section>
`;
